async-nats/src/jetstream/stream.rs

```diff
@@ -1065,6 +1065,24 @@ pub struct Config {
     /// Allow applying a subject transform to incoming messages
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub subject_transform: Option<SubjectTransform>,
+
+    #[cfg(feature = "server_2_10")]
+    /// Override compression config for this stream.
+    /// Wrapping enum that has `None` type with [Option] is there
+    /// because [Stream] can override global compression set to [Compression::S2]
+    /// to [Compression::None], which is different from not overriding global config with anything.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub compression: Option<Compression>,
```

Comment:
I'm trying to figure out how to play with compression while the docs haven't been updated yet and reading this doc comment made me think compression could be enabled globally on the server,  which is not how it seems to actually work by looking at the implementation.

I'm now confused about why we have 3 options (`Some(Compression::None)`, `Some(Compression::S2)` and `None`) instead of just `Some(Compression::S2)` and `None` or `Compression::None` and `Compression::S2`